### PR TITLE
feat: skip training in UI if valid file is present from previous runs

### PR DESCRIPTION
**Pull request by shekhar-rudder**
## Description of the change

When running the project from UI, the artefacts get uploaded on S3. Updated the training check method to check for files in S3 if the details are provided in site config

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by dpatchigolla — APPROVED** [review]

File: src/predictions/profiles_mlcorelib/py_native/training.py [hunk 1]
```diff
@@ -95,9 +96,28 @@ def describe(self, this: WhtMaterial):
     def _skip_training(self, this: WhtMaterial):
         if not self.build_spec.get("validity_time"):
             return False
-        validity_time = self.build_spec.get("validity_time")
+        site_config_path = this.wht_ctx.site_config().get("FilePath")
```
[the remark this answers is not in the document]
Comment (shekhar-rudder) [reply src/predictions/profiles_mlcorelib/py_native/training.py:99]: I didn't get this comment. Can you point out where `filepath` as a key is being used for getting the site config path ?

[the remark this answers is not in the document]
Comment (dpatchigolla) [reply src/predictions/profiles_mlcorelib/py_native/training.py:99]: Not in the code. I was looking at the siteconfig file in my local machine and I have `filepath` key there. It's not being used anywhere. 
I just mentioned it here so you would check the case-sensitivity is correct.

[the remark this answers is not in the document]
Comment (shekhar-rudder) [reply src/predictions/profiles_mlcorelib/py_native/training.py:99]: Ok. Yes in the code it becomes `FilePath`

File: src/predictions/profiles_mlcorelib/utils/S3Utils.py [hunk 2]
```diff
@@ -34,6 +35,44 @@ def delete_directory(bucket_name, aws_region_name, folder_name):
...
+        prefix,
+        download_directory,
+        folder_substring,
+        min_creation_time,
+    ):
+        s3 = boto3.client("s3", region_name=region)
+        response = s3.list_objects_v2(Bucket=bucket_name, Prefix=prefix)
+        for obj in response.get("Contents", []):
+            creation_time = obj["LastModified"]
+            isRecipeFile = folder_substring in os.path.basename(
```
[the remark this answers is not in the document]
Comment (shekhar-rudder) [reply src/predictions/profiles_mlcorelib/utils/S3Utils.py]: Updated

**Pull request merged**